Sample/SwiftUI/UserInfoView.swift

```diff
@@ -52,26 +53,34 @@ struct UserInfoView: View {
                 guard validateInput() else {
                     return
                 }
-                let userInfo = UserInfo(
-                    userID: userIDTextFieldText,
-                    idTrackingIdentifier: idTrackerTextFieldText,
-                    accessToken: accessTokenuserIDTextFieldText
-                )
-                RInAppMessaging.registerPreference(userInfo)
+                if userInfo == nil {
+                    updateUserInfoValue()
+                    RInAppMessaging.registerPreference(userInfo!)
```

Comment:
please move this code into init method. That way you won't have to check for nil and have non-optional `userInfo`